x-pack/legacy/plugins/ml/server/routes/annotations.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Boom from 'boom';
+import _ from 'lodash';
+import { i18n } from '@kbn/i18n';
+
+import { schema } from '@kbn/config-schema';
+import { isAnnotationsFeatureAvailable } from '../lib/check_annotations';
+import { annotationServiceProvider } from '../models/annotation_service';
+import { wrapError } from '../client/error_wrapper';
+import { licensePreRoutingFactory } from '../new_platform/licence_check_pre_routing_factory';
+import { RouteInitialization } from '../new_platform/plugin';
+import { indexAnnotationSchema } from '../new_platform/annotations_schema';
+
+import { ANNOTATION_USER_UNKNOWN } from '../../common/constants/annotations';
+
+function getAnnotationsFeatureUnavailableErrorMessage() {
+  return Boom.badRequest(
+    i18n.translate('xpack.ml.routes.annotations.annotationsFeatureUnavailableErrorMessage', {
+      defaultMessage:
+        'Index and aliases required for the annotations feature have not been' +
+        ' created or are not accessible for the current user.',
+    })
+  );
+}
+/**
+ * Routes for annotations
+ */
+export function annotationRoutes({ xpackMainPlugin, router }: RouteInitialization) {
+  /**
+   * @apiGroup Annotations
+   *
+   * @api {post} /api/ml/annotations Gets annotations
+   * @apiName GetAnnotations
+   * @apiDescription Gets annotations.
+   *
+   * @apiParam {String[]} jobIds List of job IDs
+   * @apiParam {String} earliestMs
+   * @apiParam {Number} latestMs
+   * @apiParam {Number} maxAnnotations Max limit of annotations returned
+   *
+   * @apiSuccess {Boolean} success
+   * @apiSuccess {Object} annotations
+   */
+  router.post(
+    {
+      path: '/api/ml/annotations',
+      validate: {
+        body: schema.object({
```

Comment:
What do you think of moving all schema definitions to `new_platform/annotations_schema.ts` for consistency even if they are not as complex as the `indexAnnotationSchema` one?